From 3cb4688b0b353296724b75772725e80fe2486958 Mon Sep 17 00:00:00 2001
From: Sebastian Ott <sebott@redhat.com>
Date: Fri, 19 Apr 2024 17:20:49 +0200
Subject: [PATCH 7/9] remove stale compat definitions

RH-Author: Sebastian Ott <sebott@redhat.com>
RH-MergeRequest: 270: RHEL10 machine types
RH-Jira: RHEL-29002 RHEL-29003 RHEL-35587 RHEL-38411 RHEL-45141 RHEL-52318 RHEL-52320
RH-Acked-by: Thomas Huth <thuth@redhat.com>
RH-Acked-by: Cornelia Huck <cohuck@redhat.com>
RH-Acked-by: Eric Auger <eric.auger@redhat.com>
RH-Commit: [7/7] b8dd0980c40974787ed80ccd216c299f0a1aefb7 (seott1/cos-qemu-kvm)

Get rid of unused [pc|hw_compat]_rhel_[7|8]* definitions.

Signed-off-by: Sebastian Ott <sebott@redhat.com>
---
 hw/core/machine.c    | 196 -------------------------------------------
 hw/i386/pc.c         | 115 -------------------------
 include/hw/boards.h  |  24 ------
 include/hw/i386/pc.h |  21 -----
 4 files changed, 356 deletions(-)

diff --git a/hw/core/machine.c b/hw/core/machine.c
index 9cf8242b32..d95f246f66 100644
--- a/hw/core/machine.c
+++ b/hw/core/machine.c
@@ -407,202 +407,6 @@ GlobalProperty hw_compat_rhel_9_0[] = {
 };
 const size_t hw_compat_rhel_9_0_len = G_N_ELEMENTS(hw_compat_rhel_9_0);
 
-GlobalProperty hw_compat_rhel_8_6[] = {
-    /* hw_compat_rhel_8_6 bz 2065589 */
-    /*
-     * vhost-vsock device in RHEL 8 kernels doesn't support seqpacket, so
-     * we need do disable it downstream on the latest hw_compat_rhel_8.
-     */
-    { "vhost-vsock-device", "seqpacket", "off" },
-};
-const size_t hw_compat_rhel_8_6_len = G_N_ELEMENTS(hw_compat_rhel_8_6);
-
-/*
- * Mostly the same as hw_compat_6_0 and hw_compat_6_1
- */
-GlobalProperty hw_compat_rhel_8_5[] = {
-    /* hw_compat_rhel_8_5 from hw_compat_6_0 */
-    { "gpex-pcihost", "allow-unmapped-accesses", "false" },
-    /* hw_compat_rhel_8_5 from hw_compat_6_0 */
-    { "i8042", "extended-state", "false"},
-    /* hw_compat_rhel_8_5 from hw_compat_6_0 */
-    { "nvme-ns", "eui64-default", "off"},
-    /* hw_compat_rhel_8_5 from hw_compat_6_0 */
-    { "e1000", "init-vet", "off" },
-    /* hw_compat_rhel_8_5 from hw_compat_6_0 */
-    { "e1000e", "init-vet", "off" },
-    /* hw_compat_rhel_8_5 from hw_compat_6_0 */
-    { "vhost-vsock-device", "seqpacket", "off" },
-    /* hw_compat_rhel_8_5 from hw_compat_6_1 */
-    { "vhost-user-vsock-device", "seqpacket", "off" },
-    /* hw_compat_rhel_8_5 from hw_compat_6_1 */
-    { "nvme-ns", "shared", "off" },
-};
-const size_t hw_compat_rhel_8_5_len = G_N_ELEMENTS(hw_compat_rhel_8_5);
-
-/*
- * Mostly the same as hw_compat_5_2
- */
-GlobalProperty hw_compat_rhel_8_4[] = {
-    /* hw_compat_rhel_8_4 from hw_compat_5_2 */
-    { "ICH9-LPC", "smm-compat", "on"},
-    /* hw_compat_rhel_8_4 from hw_compat_5_2 */
-    { "PIIX4_PM", "smm-compat", "on"},
-    /* hw_compat_rhel_8_4 from hw_compat_5_2 */
-    { "virtio-blk-device", "report-discard-granularity", "off" },
-    /* hw_compat_rhel_8_4 from hw_compat_5_2 */
-    /*
-     * Upstream incorrectly had "virtio-net-pci" instead of "virtio-net-pci-base",
-     * (https://bugzilla.redhat.com/show_bug.cgi?id=1999141)
-     */
-    { "virtio-net-pci-base", "vectors", "3"},
-};
-const size_t hw_compat_rhel_8_4_len = G_N_ELEMENTS(hw_compat_rhel_8_4);
-
-/*
- * Mostly the same as hw_compat_5_1
- */
-GlobalProperty hw_compat_rhel_8_3[] = {
-    /* hw_compat_rhel_8_3 from hw_compat_5_1 */
-    { "vhost-scsi", "num_queues", "1"},
-    /* hw_compat_rhel_8_3 from hw_compat_5_1 */
-    { "vhost-user-blk", "num-queues", "1"},
-    /* hw_compat_rhel_8_3 from hw_compat_5_1 */
-    { "vhost-user-scsi", "num_queues", "1"},
-    /* hw_compat_rhel_8_3 from hw_compat_5_1 */
-    { "virtio-blk-device", "num-queues", "1"},
-    /* hw_compat_rhel_8_3 from hw_compat_5_1 */
-    { "virtio-scsi-device", "num_queues", "1"},
-    /* hw_compat_rhel_8_3 from hw_compat_5_1 */
-    { "nvme", "use-intel-id", "on"},
-    /* hw_compat_rhel_8_3 from hw_compat_5_1 */
-    { "pvpanic", "events", "1"}, /* PVPANIC_PANICKED */
-    /* hw_compat_rhel_8_3 from hw_compat_5_1 */
-    { "pl011", "migrate-clk", "off" },
-    /* hw_compat_rhel_8_3 bz 1912846 */
-    { "pci-xhci", "x-rh-late-msi-cap", "off" },
-    /* hw_compat_rhel_8_3 from hw_compat_5_1 */
-    { "virtio-pci", "x-ats-page-aligned", "off"},
-};
-const size_t hw_compat_rhel_8_3_len = G_N_ELEMENTS(hw_compat_rhel_8_3);
-
-/*
- * The same as hw_compat_4_2 + hw_compat_5_0
- */
-GlobalProperty hw_compat_rhel_8_2[] = {
-    /* hw_compat_rhel_8_2 from hw_compat_4_2 */
-    { "virtio-blk-device", "queue-size", "128"},
-    /* hw_compat_rhel_8_2 from hw_compat_4_2 */
-    { "virtio-scsi-device", "virtqueue_size", "128"},
-    /* hw_compat_rhel_8_2 from hw_compat_4_2 */
-    { "virtio-blk-device", "x-enable-wce-if-config-wce", "off" },
-    /* hw_compat_rhel_8_2 from hw_compat_4_2 */
-    { "virtio-blk-device", "seg-max-adjust", "off"},
-    /* hw_compat_rhel_8_2 from hw_compat_4_2 */
-    { "virtio-scsi-device", "seg_max_adjust", "off"},
-    /* hw_compat_rhel_8_2 from hw_compat_4_2 */
-    { "vhost-blk-device", "seg_max_adjust", "off"},
-    /* hw_compat_rhel_8_2 from hw_compat_4_2 */
-    { "usb-host", "suppress-remote-wake", "off" },
-    /* hw_compat_rhel_8_2 from hw_compat_4_2 */
-    { "usb-redir", "suppress-remote-wake", "off" },
-    /* hw_compat_rhel_8_2 from hw_compat_4_2 */
-    { "qxl", "revision", "4" },
-    /* hw_compat_rhel_8_2 from hw_compat_4_2 */
-    { "qxl-vga", "revision", "4" },
-    /* hw_compat_rhel_8_2 from hw_compat_4_2 */
-    { "fw_cfg", "acpi-mr-restore", "false" },
-    /* hw_compat_rhel_8_2 from hw_compat_4_2 */
-    { "virtio-device", "use-disabled-flag", "false" },
-    /* hw_compat_rhel_8_2 from hw_compat_5_0 */
-    { "pci-host-bridge", "x-config-reg-migration-enabled", "off" },
-    /* hw_compat_rhel_8_2 from hw_compat_5_0 */
-    { "virtio-balloon-device", "page-poison", "false" },
-    /* hw_compat_rhel_8_2 from hw_compat_5_0 */
-    { "vmport", "x-read-set-eax", "off" },
-    /* hw_compat_rhel_8_2 from hw_compat_5_0 */
-    { "vmport", "x-signal-unsupported-cmd", "off" },
-    /* hw_compat_rhel_8_2 from hw_compat_5_0 */
-    { "vmport", "x-report-vmx-type", "off" },
-    /* hw_compat_rhel_8_2 from hw_compat_5_0 */
-    { "vmport", "x-cmds-v2", "off" },
-    /* hw_compat_rhel_8_2 from hw_compat_5_0 */
-    { "virtio-device", "x-disable-legacy-check", "true" },
-};
-const size_t hw_compat_rhel_8_2_len = G_N_ELEMENTS(hw_compat_rhel_8_2);
-
-/*
- * The same as hw_compat_4_1
- */
-GlobalProperty hw_compat_rhel_8_1[] = {
-    /* hw_compat_rhel_8_1 from hw_compat_4_1 */
-    { "virtio-pci", "x-pcie-flr-init", "off" },
-};
-const size_t hw_compat_rhel_8_1_len = G_N_ELEMENTS(hw_compat_rhel_8_1);
-
-/* The same as hw_compat_3_1
- * format of array has been changed by:
- *     6c36bddf5340 ("machine: Use shorter format for GlobalProperty arrays")
- */
-GlobalProperty hw_compat_rhel_8_0[] = {
-    /* hw_compat_rhel_8_0 from hw_compat_3_1 */
-    { "pcie-root-port", "x-speed", "2_5" },
-    /* hw_compat_rhel_8_0 from hw_compat_3_1 */
-    { "pcie-root-port", "x-width", "1" },
-    /* hw_compat_rhel_8_0 from hw_compat_3_1 */
-    { "memory-backend-file", "x-use-canonical-path-for-ramblock-id", "true" },
-    /* hw_compat_rhel_8_0 from hw_compat_3_1 */
-    { "memory-backend-memfd", "x-use-canonical-path-for-ramblock-id", "true" },
-    /* hw_compat_rhel_8_0 from hw_compat_3_1 */
-    { "tpm-crb", "ppi", "false" },
-    /* hw_compat_rhel_8_0 from hw_compat_3_1 */
-    { "tpm-tis", "ppi", "false" },
-    /* hw_compat_rhel_8_0 from hw_compat_3_1 */
-    { "usb-kbd", "serial", "42" },
-    /* hw_compat_rhel_8_0 from hw_compat_3_1 */
-    { "usb-mouse", "serial", "42" },
-    /* hw_compat_rhel_8_0 from hw_compat_3_1 */
-    { "usb-tablet", "serial", "42" },
-    /* hw_compat_rhel_8_0 from hw_compat_3_1 */
-    { "virtio-blk-device", "discard", "false" },
-    /* hw_compat_rhel_8_0 from hw_compat_3_1 */
-    { "virtio-blk-device", "write-zeroes", "false" },
-    /* hw_compat_rhel_8_0 from hw_compat_4_0 */
-    { "VGA",            "edid", "false" },
-    /* hw_compat_rhel_8_0 from hw_compat_4_0 */
-    { "secondary-vga",  "edid", "false" },
-    /* hw_compat_rhel_8_0 from hw_compat_4_0 */
-    { "bochs-display",  "edid", "false" },
-    /* hw_compat_rhel_8_0 from hw_compat_4_0 */
-    { "virtio-vga",     "edid", "false" },
-    /* hw_compat_rhel_8_0 from hw_compat_4_0 */
-    { "virtio-gpu-device", "edid", "false" },
-    /* hw_compat_rhel_8_0 from hw_compat_4_0 */
-    { "virtio-device", "use-started", "false" },
-    /* hw_compat_rhel_8_0 from hw_compat_3_1 - that was added in 4.1 */
-    { "pcie-root-port-base", "disable-acs", "true" },
-};
-const size_t hw_compat_rhel_8_0_len = G_N_ELEMENTS(hw_compat_rhel_8_0);
-
-/* The same as hw_compat_3_0 + hw_compat_2_12
- * except that
- *   there's nothing in 3_0
- *   migration.decompress-error-check=off was in 7.5 from bz 1584139
- */
-GlobalProperty hw_compat_rhel_7_6[] = {
-    /* hw_compat_rhel_7_6 from hw_compat_2_12 */
-    { "hda-audio", "use-timer", "false" },
-    /* hw_compat_rhel_7_6 from hw_compat_2_12 */
-    { "cirrus-vga", "global-vmstate", "true" },
-    /* hw_compat_rhel_7_6 from hw_compat_2_12 */
-    { "VGA", "global-vmstate", "true" },
-    /* hw_compat_rhel_7_6 from hw_compat_2_12 */
-    { "vmware-svga", "global-vmstate", "true" },
-    /* hw_compat_rhel_7_6 from hw_compat_2_12 */
-    { "qxl-vga", "global-vmstate",  "true" },
-};
-const size_t hw_compat_rhel_7_6_len = G_N_ELEMENTS(hw_compat_rhel_7_6);
-
 MachineState *current_machine;
 
 static char *machine_get_kernel(Object *obj, Error **errp)
diff --git a/hw/i386/pc.c b/hw/i386/pc.c
index a49d346d2e..7af762065f 100644
--- a/hw/i386/pc.c
+++ b/hw/i386/pc.c
@@ -316,121 +316,6 @@ GlobalProperty pc_rhel_9_0_compat[] = {
 };
 const size_t pc_rhel_9_0_compat_len = G_N_ELEMENTS(pc_rhel_9_0_compat);
 
-GlobalProperty pc_rhel_8_5_compat[] = {
-    /* pc_rhel_8_5_compat from pc_compat_6_0 */
-    { "qemu64" "-" TYPE_X86_CPU, "family", "6" },
-    /* pc_rhel_8_5_compat from pc_compat_6_0 */
-    { "qemu64" "-" TYPE_X86_CPU, "model", "6" },
-    /* pc_rhel_8_5_compat from pc_compat_6_0 */
-    { "qemu64" "-" TYPE_X86_CPU, "stepping", "3" },
-    /* pc_rhel_8_5_compat from pc_compat_6_0 */
-    { TYPE_X86_CPU, "x-vendor-cpuid-only", "off" },
-    /* pc_rhel_8_5_compat from pc_compat_6_0 */
-    { "ICH9-LPC", ACPI_PM_PROP_ACPI_PCIHP_BRIDGE, "off" },
-
-    /* pc_rhel_8_5_compat from pc_compat_6_1 */
-    { TYPE_X86_CPU, "hv-version-id-build", "0x1bbc" },
-    /* pc_rhel_8_5_compat from pc_compat_6_1 */
-    { TYPE_X86_CPU, "hv-version-id-major", "0x0006" },
-    /* pc_rhel_8_5_compat from pc_compat_6_1 */
-    { TYPE_X86_CPU, "hv-version-id-minor", "0x0001" },
-};
-const size_t pc_rhel_8_5_compat_len = G_N_ELEMENTS(pc_rhel_8_5_compat);
-
-GlobalProperty pc_rhel_8_4_compat[] = {
-    /* pc_rhel_8_4_compat from pc_compat_5_2 */
-    { "ICH9-LPC", "x-smi-cpu-hotunplug", "off" },
-    { TYPE_X86_CPU, "kvm-asyncpf-int", "off" },
-};
-const size_t pc_rhel_8_4_compat_len = G_N_ELEMENTS(pc_rhel_8_4_compat);
-
-GlobalProperty pc_rhel_8_3_compat[] = {
-    /* pc_rhel_8_3_compat from pc_compat_5_1 */
-    { "ICH9-LPC", "x-smi-cpu-hotplug", "off" },
-};
-const size_t pc_rhel_8_3_compat_len = G_N_ELEMENTS(pc_rhel_8_3_compat);
-
-GlobalProperty pc_rhel_8_2_compat[] = {
-    /* pc_rhel_8_2_compat from pc_compat_4_2 */
-    { "mch", "smbase-smram", "off" },
-};
-const size_t pc_rhel_8_2_compat_len = G_N_ELEMENTS(pc_rhel_8_2_compat);
-
-/* pc_rhel_8_1_compat is empty since pc_4_1_compat is */
-GlobalProperty pc_rhel_8_1_compat[] = { };
-const size_t pc_rhel_8_1_compat_len = G_N_ELEMENTS(pc_rhel_8_1_compat);
-
-GlobalProperty pc_rhel_8_0_compat[] = {
-    /* pc_rhel_8_0_compat from pc_compat_3_1 */
-    { "intel-iommu", "dma-drain", "off" },
-    /* pc_rhel_8_0_compat from pc_compat_3_1 */
-    { "Opteron_G3" "-" TYPE_X86_CPU, "rdtscp", "off" },
-    /* pc_rhel_8_0_compat from pc_compat_3_1 */
-    { "Opteron_G4" "-" TYPE_X86_CPU, "rdtscp", "off" },
-    /* pc_rhel_8_0_compat from pc_compat_3_1 */
-    { "Opteron_G4" "-" TYPE_X86_CPU, "npt", "off" },
-    /* pc_rhel_8_0_compat from pc_compat_3_1 */
-    { "Opteron_G4" "-" TYPE_X86_CPU, "nrip-save", "off" },
-    /* pc_rhel_8_0_compat from pc_compat_3_1 */
-    { "Opteron_G5" "-" TYPE_X86_CPU, "rdtscp", "off" },
-    /* pc_rhel_8_0_compat from pc_compat_3_1 */
-    { "Opteron_G5" "-" TYPE_X86_CPU, "npt", "off" },
-    /* pc_rhel_8_0_compat from pc_compat_3_1 */
-    { "Opteron_G5" "-" TYPE_X86_CPU, "nrip-save", "off" },
-    /* pc_rhel_8_0_compat from pc_compat_3_1 */
-    { "EPYC" "-" TYPE_X86_CPU, "npt", "off" },
-    /* pc_rhel_8_0_compat from pc_compat_3_1 */
-    { "EPYC" "-" TYPE_X86_CPU, "nrip-save", "off" },
-    /* pc_rhel_8_0_compat from pc_compat_3_1 */
-    { "EPYC-IBPB" "-" TYPE_X86_CPU, "npt", "off" },
-    /* pc_rhel_8_0_compat from pc_compat_3_1 */
-    { "EPYC-IBPB" "-" TYPE_X86_CPU, "nrip-save", "off" },
-    /** The mpx=on entries from pc_compat_3_1 are in pc_rhel_7_6_compat **/
-    /* pc_rhel_8_0_compat from pc_compat_3_1 */
-    { "Cascadelake-Server" "-" TYPE_X86_CPU, "stepping", "5" },
-    /* pc_rhel_8_0_compat from pc_compat_3_1 */
-    { TYPE_X86_CPU, "x-intel-pt-auto-level", "off" },
-};
-const size_t pc_rhel_8_0_compat_len = G_N_ELEMENTS(pc_rhel_8_0_compat);
-
-/* Similar to PC_COMPAT_3_0 + PC_COMPAT_2_12, but:
- * all of the 2_12 stuff was already in 7.6 from bz 1481253
- * x-migrate-smi-count comes from PC_COMPAT_2_11 but
- * is really tied to kernel version so keep it off on 7.x
- * machine types irrespective of host.
- */
-GlobalProperty pc_rhel_7_6_compat[] = {
-    /* pc_rhel_7_6_compat from pc_compat_3_0 */
-    { TYPE_X86_CPU, "x-hv-synic-kvm-only", "on" },
-    /* pc_rhel_7_6_compat from pc_compat_3_0 */
-    { "Skylake-Server" "-" TYPE_X86_CPU, "pku", "off" },
-    /* pc_rhel_7_6_compat from pc_compat_3_0 */
-    { "Skylake-Server-IBRS" "-" TYPE_X86_CPU, "pku", "off" },
-    /* pc_rhel_7_6_compat from pc_compat_2_11 */
-    { TYPE_X86_CPU, "x-migrate-smi-count", "off" },
-    /* pc_rhel_7_6_compat from pc_compat_2_11 */
-    { "Skylake-Client" "-" TYPE_X86_CPU, "mpx", "on" },
-    /* pc_rhel_7_6_compat from pc_compat_2_11 */
-    { "Skylake-Client-IBRS" "-" TYPE_X86_CPU, "mpx", "on" },
-    /* pc_rhel_7_6_compat from pc_compat_2_11 */
-    { "Skylake-Server" "-" TYPE_X86_CPU, "mpx", "on" },
-    /* pc_rhel_7_6_compat from pc_compat_2_11 */
-    { "Skylake-Server-IBRS" "-" TYPE_X86_CPU, "mpx", "on" },
-    /* pc_rhel_7_6_compat from pc_compat_2_11 */
-    { "Cascadelake-Server" "-" TYPE_X86_CPU, "mpx", "on" },
-    /* pc_rhel_7_6_compat from pc_compat_2_11 */
-    { "Icelake-Client" "-" TYPE_X86_CPU, "mpx", "on" },
-    /* pc_rhel_7_6_compat from pc_compat_2_11 */
-    { "Icelake-Server" "-" TYPE_X86_CPU, "mpx", "on" },
-};
-const size_t pc_rhel_7_6_compat_len = G_N_ELEMENTS(pc_rhel_7_6_compat);
-
-/*
- * The PC_RHEL_*_COMPAT serve the same purpose for RHEL-7 machine
- * types as the PC_COMPAT_* do for upstream types.
- * PC_RHEL_7_*_COMPAT apply both to i440fx and q35 types.
- */
-
 GSIState *pc_gsi_create(qemu_irq **irqs, bool pci_enabled)
 {
     GSIState *s;
diff --git a/include/hw/boards.h b/include/hw/boards.h
index 6d98aaf4c7..ac917b87fb 100644
--- a/include/hw/boards.h
+++ b/include/hw/boards.h
@@ -823,29 +823,5 @@ extern const size_t hw_compat_rhel_9_1_len;
 extern GlobalProperty hw_compat_rhel_9_0[];
 extern const size_t hw_compat_rhel_9_0_len;
 
-extern GlobalProperty hw_compat_rhel_8_6[];
-extern const size_t hw_compat_rhel_8_6_len;
-
-extern GlobalProperty hw_compat_rhel_8_5[];
-extern const size_t hw_compat_rhel_8_5_len;
-
-extern GlobalProperty hw_compat_rhel_8_4[];
-extern const size_t hw_compat_rhel_8_4_len;
-
-extern GlobalProperty hw_compat_rhel_8_3[];
-extern const size_t hw_compat_rhel_8_3_len;
-
-extern GlobalProperty hw_compat_rhel_8_2[];
-extern const size_t hw_compat_rhel_8_2_len;
-
-extern GlobalProperty hw_compat_rhel_8_1[];
-extern const size_t hw_compat_rhel_8_1_len;
-
-extern GlobalProperty hw_compat_rhel_8_0[];
-extern const size_t hw_compat_rhel_8_0_len;
-
-extern GlobalProperty hw_compat_rhel_7_6[];
-extern const size_t hw_compat_rhel_7_6_len;
-
 extern const char *rhel_old_machine_deprecation;
 #endif
diff --git a/include/hw/i386/pc.h b/include/hw/i386/pc.h
index 8e9597f40f..61609027d0 100644
--- a/include/hw/i386/pc.h
+++ b/include/hw/i386/pc.h
@@ -314,27 +314,6 @@ extern const size_t pc_rhel_9_2_compat_len;
 extern GlobalProperty pc_rhel_9_0_compat[];
 extern const size_t pc_rhel_9_0_compat_len;
 
-extern GlobalProperty pc_rhel_8_5_compat[];
-extern const size_t pc_rhel_8_5_compat_len;
-
-extern GlobalProperty pc_rhel_8_4_compat[];
-extern const size_t pc_rhel_8_4_compat_len;
-
-extern GlobalProperty pc_rhel_8_3_compat[];
-extern const size_t pc_rhel_8_3_compat_len;
-
-extern GlobalProperty pc_rhel_8_2_compat[];
-extern const size_t pc_rhel_8_2_compat_len;
-
-extern GlobalProperty pc_rhel_8_1_compat[];
-extern const size_t pc_rhel_8_1_compat_len;
-
-extern GlobalProperty pc_rhel_8_0_compat[];
-extern const size_t pc_rhel_8_0_compat_len;
-
-extern GlobalProperty pc_rhel_7_6_compat[];
-extern const size_t pc_rhel_7_6_compat_len;
-
 #define DEFINE_PC_MACHINE(suffix, namestr, initfn, optsfn) \
     static void pc_machine_##suffix##_class_init(ObjectClass *oc, void *data) \
     { \
-- 
2.39.3

